wisp-tracing/src/main/kotlin/wisp/tracing/TracerExt.kt

```diff
@@ -1,38 +1,51 @@
 package wisp.tracing
 
+import io.opentracing.Scope
 import io.opentracing.Span
 import io.opentracing.Tracer
 import io.opentracing.tag.Tags
 
-/** [trace] traces the given function with the specific span name and optional tags */
+/**
+ * Traces a function [f], using a span called [spanName], which is automatically finished when the
+ * function completes execution.
+ *
+ * If a span is already active, the new span is made a child of the existing one.
+ * If you want to manipulate the [Span] (e.g. to attach baggage), use [traceWithSpan] instead.
+ *
+ * If you want a new independent span, use [traceWithNewRootSpan].
+ */
 fun <T : Any?> Tracer.trace(spanName: String, tags: Map<String, String> = mapOf(), f: () -> T): T =
```

Comment:
could we do `block` instead of `f` as the unit parameter name in these methods?

Reply:
We could, and I share that preference, but I didn't want to rename the parameter because its a very tiny nit, and possibly an addition breaking change.